beacon_node/http_api/src/lib.rs

```diff
@@ -1999,11 +2000,14 @@ pub fn serve<T: BeaconChainTypes>(
              chain: Arc<BeaconChain<T>>,
              query: api_types::AttestationPoolQuery| {
                 task_spawner.blocking_response_task(Priority::P1, move || {
-                    let query_filter = |data: &AttestationData| {
+                    let query_filter = |data: &AttestationData, committee_index: Option<u64>| {
                         query.slot.is_none_or(|slot| slot == data.slot)
-                            && query
-                                .committee_index
-                                .is_none_or(|index| index == data.index)
+                            && query.committee_index.is_none_or(|index| {
+                                if let Some(committee_index) = committee_index {
+                                    return index == committee_index;
+                                }
+                                false
```

Comment:
this is `committee_index.is_some_and(|committee_index| index == committee_index)`